Parse text

labouz opened the pull request:
needs tests

labouz committed:
beginning of parse func

labouz committed:
got prompt to work and func to subset to just drug col

labouz committed:
unnested tokens, regex to clean names

labouz committed:
test on lookup

labouz committed:
write parse into dope

labouz committed:
change first arg to data object not path; issue #8

labouz committed:
added special cases to parse(), created test data; issue #7

labouz committed:
fixed data note

labouz committed:
update deescription, remove test file

labouz committed:
change arg to vector instead of df; issue #8

labouz committed:
add more drug stop-words based on iqvia, modify bup/nal combinations; closes issue #7; CTN94-117

labouz committed:
tweak parse based on iqvia, adding classes and categories; CTN94-102

labouz committed:
add categories and classes to iqvia data

labouz committed:
bind iqvia to lookup; CTN94-102

labouz committed:
Merge pull request #16 from CTN-0094/iqvia

Iqvia;  CTN94-102

labouz commented:
@nathan-castor this function needs a test pls

nathan-castor commented:
@labouz 
Looks like there are some conflicts with this code now, sorry I would've done this last week if I could have.

labouz committed:
manage lookup class inconsistencies

labouz committed:
Merge branch 'iqvia' into parse_text

nathan-castor commented on R/parse.R:
```diff
@@ -0,0 +1,109 @@
+  #   colNum <- as.numeric(readline(prompt))
+  #   drug_data[ ,colNum] #returns tibble of n X 1
+  # }
+
```
Let's remove the commented code that's not serving a purpose

nathan-castor commented on R/parse.R:
```diff
@@ -0,0 +1,109 @@
+#'
+#' @examples
+#' \dontrun{
+#' parse(drug_df)
```
Incorrect argument name (drug_vec)

labouz committed:
unignore drug_df.rda

nathan-castor commented on R/parse.R:
```diff
@@ -0,0 +1,109 @@
+parse <- function(drug_vec){
+
+  #binding vars to function
+  word <- drug <- for_token <- stop_words <- NULL
```
we don't need stop_words set to NULL here

nathan-castor commented on R/parse.R:
```diff
@@ -0,0 +1,109 @@
+#' Parse a vector of free text containing drug information
+#'
+#' @description This function provides a dataframe of parsed out strings from a
+#'   free text field given a file path specified by the user.
```
edit description: we don't take a file path anymore

nathan-castor commented on R/parse.R:
```diff
@@ -0,0 +1,109 @@
+  # drugs <- get_col()
+  #convert vector to df
+  drugs <- as.data.frame(drug_vec)
+  drugCol <- as.name(names(drugs))
```
unnecessary to create this variable because you are piping the df into the function on line 67

nathan-castor reviewed (changes requested):
Woohoo!

labouz committed:
making changes requested by @nathan-castor; resolves issue #8 & #7; CTN94-101; CTN94-117

nathan-castor commented on R/parse.R:
```diff
@@ -0,0 +1,90 @@
+                       "intensol", "ject", "calcium")
+
+  ##INTERACTIVE - must respond!
+  # drugs <- get_col()
```
unneeded comments (commented code)

nathan-castor commented on R/parse.R:
```diff
@@ -0,0 +1,90 @@
+  # drugs <- get_col()
+  #convert vector to df
+  drugs <- as.data.frame(drug_vec)
+  # drugCol <- as.name(names(drugs))
```
unneeded comments (commented code)

nathan-castor reviewed (changes requested):
just two lines of commented code

labouz committed:
remove comments as per @nathaniel-castor

labouz committed:
address conflicts

Merge branch 'master' into parse_text

# Conflicts:
#	.gitignore
#	data/lookup_df.rda
